p2p/errors_internal_test.go

```diff
@@ -102,3 +103,26 @@ func TestDialErrMsgs(t *testing.T) {
 	require.Contains(t, msgs[badAddr.String()], "dial backoff")
 	require.Contains(t, msgs[hostBAddr.String()], "dial backoff")
 }
+
+// closedGater is a connection gater that blocks all connections.
+type closedGater struct{}
+
+func (closedGater) InterceptPeerDial(peer.ID) bool {
```

Comment:
okay so what we were doing earlier returned true even with nil, parameters which is why this test was failing